Fix linking of the test programs.

From: David Rothenberger <daveroth@acm.org>

The libsvn_test library contains undefined references, so it can't
be turned into a dynamic library. To fix this, remove that library
and modify all the tests so they include the sources in libsvn_test
directly.
---
 build.conf |  395 ++++++++++++++++++++++++++++++------------------------------
 1 file changed, 198 insertions(+), 197 deletions(-)

diff --git a/build.conf b/build.conf
index d87f66b..018dbb8 100644
--- a/build.conf
+++ b/build.conf
@@ -779,13 +779,14 @@ link-cmd = $(LINK_CXX_LIB)
 #
 
 # general library:  our C testing framework
-[libsvn_test]
-type = lib
-path = subversion/tests
-install = test
-libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr
-msvc-static = yes
-undefined-lib-symbols = yes
+#[libsvn_test]
+#type = lib
+#path = subversion/tests
+#sources = svn_test_fs.c
+#install = test
+#libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr
+#msvc-static = yes
+#undefined-lib-symbols = yes
 
 # ----------------------------------------------------------------------------
 # Tests for libsvn_fs_base
@@ -793,30 +794,30 @@ undefined-lib-symbols = yes
 [fs-base-test]
 description = Tests for *public* fs API (svn_fs.h)
 type = exe
-path = subversion/tests/libsvn_fs_base
-sources = fs-base-test.c
+path = subversion/tests
+sources = libsvn_fs_base/fs-base-test.c svn_test_fs.c svn_test_main.c
 install = bdb-test
-libs = libsvn_test libsvn_fs libsvn_fs_base libsvn_delta
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_fs libsvn_fs_base libsvn_delta
        libsvn_fs_util libsvn_subr apriconv apr
 msvc-force-static = yes
 
 [strings-reps-test]
 description = Test strings/reps in libsvn_fs_base
 type = exe
-path = subversion/tests/libsvn_fs_base
-sources = strings-reps-test.c
+path = subversion/tests
+sources = libsvn_fs_base/strings-reps-test.c svn_test_fs.c svn_test_main.c
 install = bdb-test
-libs = libsvn_test libsvn_fs libsvn_fs_base libsvn_delta
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_fs libsvn_fs_base libsvn_delta
        libsvn_subr apriconv apr
 msvc-force-static = yes
 
 [changes-test]
 description = Test changes in libsvn_fs_base
 type = exe
-path = subversion/tests/libsvn_fs_base
-sources = changes-test.c
+path = subversion/tests
+sources = libsvn_fs_base/changes-test.c svn_test_fs.c svn_test_main.c
 install = bdb-test
-libs = libsvn_test libsvn_fs libsvn_fs_base libsvn_delta
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_fs libsvn_fs_base libsvn_delta
        libsvn_subr apriconv apr
 msvc-force-static = yes
 
@@ -825,30 +826,30 @@ msvc-force-static = yes
 [fs-fs-pack-test]
 description = Test fsfs packing in libsvn_fs_fs
 type = exe
-path = subversion/tests/libsvn_fs_fs
-sources = fs-fs-pack-test.c
+path = subversion/tests
+sources = libsvn_fs_fs/fs-fs-pack-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_fs libsvn_fs_fs libsvn_delta
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_fs libsvn_fs_fs libsvn_delta
        libsvn_subr apriconv apr
 msvc-force-static = yes
 
 [fs-fs-fuzzy-test]
 description = Use fuzzying to test FSFS corruption resilience
 type = exe
-path = subversion/tests/libsvn_fs_fs
-sources = fs-fs-fuzzy-test.c
+path = subversion/tests
+sources = libsvn_fs_fs/fs-fs-fuzzy-test.c svn_test_fs.c svn_test_main.c
 install = sub-test
-libs = libsvn_test libsvn_fs libsvn_fs_fs libsvn_delta
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_fs libsvn_fs_fs libsvn_delta
        libsvn_repos libsvn_subr apriconv apr
 msvc-force-static = yes
 
 [fs-fs-private-test]
 description = Test FSSF private API
 type = exe
-path = subversion/tests/libsvn_fs_fs
-sources = fs-fs-private-test.c
+path = subversion/tests
+sources = libsvn_fs_fs/fs-fs-private-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_fs libsvn_fs_fs libsvn_delta
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_fs libsvn_fs_fs libsvn_delta
        libsvn_repos libsvn_subr apriconv apr
 msvc-force-static = yes
 
@@ -857,20 +858,20 @@ msvc-force-static = yes
 [fs-x-pack-test]
 description = Test fsx packing in libsvn_fs_x
 type = exe
-path = subversion/tests/libsvn_fs_x
-sources = fs-x-pack-test.c
+path = subversion/tests
+sources = libsvn_fs_x/fs-x-pack-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_fs libsvn_fs_x libsvn_delta
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_fs libsvn_fs_x libsvn_delta
        libsvn_subr apriconv apr
 msvc-force-static = yes
 
 [string-table-test]
 description = Test fsfs string tables
 type = exe
-path = subversion/tests/libsvn_fs_x
-sources = string-table-test.c
+path = subversion/tests
+sources = libsvn_fs_x/string-table-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_fs_x libsvn_subr apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_fs_x libsvn_subr apr
 msvc-force-static = yes
 
 # ----------------------------------------------------------------------------
@@ -879,28 +880,28 @@ msvc-force-static = yes
 [locks-test]
 description = Test locks in libsvn_fs
 type = exe
-path = subversion/tests/libsvn_fs
-sources = locks-test.c
+path = subversion/tests
+sources = libsvn_fs/locks-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_fs libsvn_delta libsvn_subr apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_fs libsvn_delta libsvn_subr apriconv apr
 msvc-force-static = yes
 
 [fs-test]
 description = Tests in libsvn_fs
 type = exe
-path = subversion/tests/libsvn_fs
-sources = fs-test.c
+path = subversion/tests
+sources = libsvn_fs/fs-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_fs libsvn_delta
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_fs libsvn_delta
        libsvn_fs_util libsvn_subr aprutil apriconv apr
 
 [fs-sequential-test]
 description = Tests in libsvn_fs run sequentially
 type = exe
-path = subversion/tests/libsvn_fs
-sources = fs-sequential-test.c
+path = subversion/tests
+sources = libsvn_fs/fs-sequential-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_fs libsvn_delta
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_fs libsvn_delta
        libsvn_fs_util libsvn_subr aprutil apriconv apr
 
 # ----------------------------------------------------------------------------
@@ -909,26 +910,26 @@ libs = libsvn_test libsvn_fs libsvn_delta
 [authz-test]
 description = Test authz parsing in libsvn_repos
 type = exe
-path = subversion/tests/libsvn_repos
-sources = authz-test.c
+path = subversion/tests
+sources = libsvn_subr/auth-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_repos libsvn_fs libsvn_delta libsvn_subr apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apr
 
 [repos-test]
 description = Test delta editor in libsvn_repos
 type = exe
-path = subversion/tests/libsvn_repos
-sources = repos-test.c dir-delta-editor.c
+path = subversion/tests
+sources = libsvn_repos/repos-test.c libsvn_repos/dir-delta-editor.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_repos libsvn_fs libsvn_delta libsvn_subr apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_repos libsvn_fs libsvn_delta libsvn_subr apriconv apr
 
 [dump-load-test]
 description = Test dumping/loading repositories in libsvn_repos
 type = exe
-path = subversion/tests/libsvn_repos
-sources = dump-load-test.c
+path = subversion/tests
+sources = libsvn_repos/dump-load-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_repos libsvn_fs libsvn_delta libsvn_subr apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_repos libsvn_fs libsvn_delta libsvn_subr apriconv apr
 
 # ----------------------------------------------------------------------------
 # Tests for libsvn_subr
@@ -936,260 +937,260 @@ libs = libsvn_test libsvn_repos libsvn_fs libsvn_delta libsvn_subr apriconv apr
 [auth-test]
 description = Test platform-specific auth provider access
 type = exe
-path = subversion/tests/libsvn_subr
-sources = auth-test.c
+path = subversion/tests
+sources = libsvn_subr/auth-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_subr apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apr
 
 [bit-array-test]
 description = Test packed bit arrays
 type = exe
-path = subversion/tests/libsvn_subr
-sources = bit-array-test.c
+path = subversion/tests
+sources = libsvn_subr/bit-array-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_subr apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apr
 
 [cache-test]
 description = Test in-memory cache
 type = exe
-path = subversion/tests/libsvn_subr
-sources = cache-test.c
+path = subversion/tests
+sources = libsvn_subr/cache-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_subr apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apr
 
 [checksum-test]
 description = Test checksum functions
 type = exe
-path = subversion/tests/libsvn_subr
-sources = checksum-test.c
+path = subversion/tests
+sources = libsvn_subr/checksum-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_subr apr zlib
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apr zlib
 msvc-force-static = yes
 
 [compat-test]
 description = Test compatibility functions
 type = exe
-path = subversion/tests/libsvn_subr
-sources = compat-test.c
+path = subversion/tests
+sources = libsvn_subr/compat-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_subr apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apr
 
 [config-test]
 description = Test svn_config utilities
 type = exe
-path = subversion/tests/libsvn_subr
-sources = config-test.c
+path = subversion/tests
+sources = libsvn_subr/config-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_subr apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apriconv apr
 
 [crypto-test]
 description = Test svn_crypto utilities
 type = exe
-path = subversion/tests/libsvn_subr
-sources = crypto-test.c
+path = subversion/tests
+sources = libsvn_subr/crypto-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_subr aprutil apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr aprutil apr
 msvc-force-static = yes
 
 [dirent_uri-test]
 description = Test dirent_uri library
 type = exe
-path = subversion/tests/libsvn_subr
-sources = dirent_uri-test.c
+path = subversion/tests
+sources = libsvn_subr/dirent_uri-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_subr apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apriconv apr
 
 [error-test]
 description = Test error library
 type = exe
-path = subversion/tests/libsvn_subr
-sources = error-test.c
+path = subversion/tests
+sources = libsvn_subr/error-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_subr apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apriconv apr
 
 [error-code-test]
 description = Test error library
 type = exe
-path = subversion/tests/libsvn_subr
-sources = error-code-test.c
+path = subversion/tests
+sources = libsvn_subr/error-code-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_subr apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apriconv apr
 
 [hashdump-test]
 description = Test hashfile format for props
 type = exe
-path = subversion/tests/libsvn_subr
-sources = hashdump-test.c
+path = subversion/tests
+sources = libsvn_subr/hashdump-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_subr apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apriconv apr
 
 [io-test]
 description = Test I/O Operations
 type = exe
-path = subversion/tests/libsvn_subr
-sources = io-test.c
+path = subversion/tests
+sources = libsvn_subr/io-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_subr apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apriconv apr
 
 [opt-test]
 description = Test options library
 type = exe
-path = subversion/tests/libsvn_subr
-sources = opt-test.c
+path = subversion/tests
+sources = libsvn_subr/opt-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_subr apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apr
 
 [mergeinfo-test]
 description = Test mergeinfo library
 type = exe
-path = subversion/tests/libsvn_subr
-sources = mergeinfo-test.c
+path = subversion/tests
+sources = libsvn_subr/mergeinfo-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_subr apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apr
 
 [packed-data-test]
 description = Test path library
 type = exe
-path = subversion/tests/libsvn_subr
-sources = packed-data-test.c
+path = subversion/tests
+sources = libsvn_subr/packed-data-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_subr apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apr
 
 [path-test]
 description = Test path library
 type = exe
-path = subversion/tests/libsvn_subr
-sources = path-test.c
+path = subversion/tests
+sources = libsvn_subr/path-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_subr apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apr
 
 [prefix-string-test]
 description = Test path library
 type = exe
-path = subversion/tests/libsvn_subr
-sources = prefix-string-test.c
+path = subversion/tests
+sources = libsvn_subr/prefix-string-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_subr apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apr
 
 [priority-queue-test]
 description = Test path library
 type = exe
-path = subversion/tests/libsvn_subr
-sources = priority-queue-test.c
+path = subversion/tests
+sources = libsvn_subr/priority-queue-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_subr apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apr
 
 [revision-test]
 description = Test revision library
 type = exe
-path = subversion/tests/libsvn_subr
-sources = revision-test.c
+path = subversion/tests
+sources = libsvn_subr/revision-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_subr apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apr
 
 [root-pools-test]
 description = Test time functions
 type = exe
-path = subversion/tests/libsvn_subr
-sources = root-pools-test.c
+path = subversion/tests
+sources = libsvn_subr/root-pools-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_subr apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apr
 
 [skel-test]
 description = Test skels in libsvn_subr
 type = exe
-path = subversion/tests/libsvn_subr
-sources = skel-test.c
+path = subversion/tests
+sources = libsvn_subr/skel-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_subr apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apr
 
 [spillbuf-test]
 description = Test spillbuf in libsvn_subr
 type = exe
-path = subversion/tests/libsvn_subr
-sources = spillbuf-test.c
+path = subversion/tests
+sources = libsvn_subr/spillbuf-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_subr apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apr
 
 [stream-test]
 description = Test stream library
 type = exe
-path = subversion/tests/libsvn_subr
-sources = stream-test.c
+path = subversion/tests
+sources = libsvn_subr/stream-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_subr apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apr
 
 [string-test]
 description = Test svn_stringbuf_t utilities
 type = exe
-path = subversion/tests/libsvn_subr
-sources = string-test.c
+path = subversion/tests
+sources = libsvn_subr/string-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_subr apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apr
 
 [sqlite-test]
 description = Test stream library
 type = exe
-path = subversion/tests/libsvn_subr
-sources = sqlite-test.c
+path = subversion/tests
+sources = libsvn_subr/sqlite-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_subr apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apr
 
 [time-test]
 description = Test time functions
 type = exe
-path = subversion/tests/libsvn_subr
-sources = time-test.c
+path = subversion/tests
+sources = libsvn_subr/time-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_subr apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apr
 
 [utf-test]
 description = Test UTF-8 functions
 type = exe
-path = subversion/tests/libsvn_subr
-sources = utf-test.c
+path = subversion/tests
+sources = libsvn_subr/utf-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_subr apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apr
 
 [subst_translate-test]
 description = Test the svn_subst_translate* functions
 type = exe
-path = subversion/tests/libsvn_subr
-sources = subst_translate-test.c
+path = subversion/tests
+sources = libsvn_subr/subst_translate-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_subr apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apr
 
 [translate-test]
 description = Test eol conversion and keyword substitution routines
 type = exe
-path = subversion/tests/libsvn_subr
-sources = translate-test.c
+path = subversion/tests
+sources = libsvn_subr/translate-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_subr apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apr
 
 [x509-test]
 description = Test x509 parser
 type = exe
-path = subversion/tests/libsvn_subr
-sources = x509-test.c
+path = subversion/tests
+sources = libsvn_subr/x509-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_subr apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apr
 
 [xml-test]
 description = Test XML parser in libsvn_subr
 type = exe
-path = subversion/tests/libsvn_subr
-sources = xml-test.c
+path = subversion/tests
+sources = libsvn_subr/xml-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_subr apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apr
 
 [compress-test]
 description = Test compression functions
 type = exe
-path = subversion/tests/libsvn_subr
-sources = compress-test.c
+path = subversion/tests
+sources = libsvn_subr/compress-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_subr apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apr
 
 # ----------------------------------------------------------------------------
 # Tests for libsvn_delta
@@ -1197,26 +1198,26 @@ libs = libsvn_test libsvn_subr apr
 [random-test]
 description = Use random data to test delta processing
 type = exe
-path = subversion/tests/libsvn_delta
-sources = random-test.c
+path = subversion/tests
+sources = libsvn_delta/random-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_delta libsvn_subr apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apr
 
 [window-test]
 description = Test delta window generation
 type = exe
-path = subversion/tests/libsvn_delta
-sources = window-test.c
+path = subversion/tests
+sources = libsvn_delta/window-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_delta libsvn_subr apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apr
 
 [svndiff-stream-test]
 description = Test svndiff streams
 type = exe
-path = subversion/tests/libsvn_delta
-sources = svndiff-stream-test.c
+path = subversion/tests
+sources = libsvn_delta/svndiff-stream-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_delta libsvn_subr apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apr
 
 # ----------------------------------------------------------------------------
 # Tests for libsvn_client
@@ -1224,27 +1225,27 @@ libs = libsvn_test libsvn_delta libsvn_subr apriconv apr
 [client-test]
 description = Test low-level functionality in libsvn_client
 type = exe
-path = subversion/tests/libsvn_client
-sources = client-test.c
+path = subversion/tests
+sources = libsvn_client/client-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_client libsvn_wc libsvn_repos libsvn_ra libsvn_fs libsvn_delta libsvn_subr apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_client libsvn_wc libsvn_repos libsvn_ra libsvn_fs libsvn_delta libsvn_subr apriconv apr
 msvc-force-static = yes
 
 [mtcc-test]
 description = Test Multi Command Context
 type = exe
-path = subversion/tests/libsvn_client
-sources = mtcc-test.c
+path = subversion/tests
+sources = libsvn_client/mtcc-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_client libsvn_wc libsvn_repos libsvn_ra libsvn_fs libsvn_delta libsvn_subr apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_client libsvn_wc libsvn_repos libsvn_ra libsvn_fs libsvn_delta libsvn_subr apriconv apr
 
 [conflicts-test]
 description = Test libsvn_client conflict resolver
 type = exe
-path = subversion/tests/libsvn_client
-sources = conflicts-test.c ../libsvn_wc/utils.c
+path = subversion/tests
+sources = libsvn_client/conflicts-test.c libsvn_wc/utils.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_client libsvn_wc libsvn_repos libsvn_ra libsvn_fs libsvn_delta libsvn_subr apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_client libsvn_wc libsvn_repos libsvn_ra libsvn_fs libsvn_delta libsvn_subr apriconv apr
 msvc-force-static = yes
 
 # ----------------------------------------------------------------------------
@@ -1253,18 +1254,18 @@ msvc-force-static = yes
 [diff-diff3-test]
 description = Test the diff/diff3 library
 type = exe
-path = subversion/tests/libsvn_diff
-sources = diff-diff3-test.c
+path = subversion/tests
+sources = libsvn_diff/diff-diff3-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_diff libsvn_subr apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_diff libsvn_subr apriconv apr
 
 [parse-diff-test]
 description = Test unidiff parsing
 type = exe
-path = subversion/tests/libsvn_diff
-sources = parse-diff-test.c
+path = subversion/tests
+sources = libsvn_diff/parse-diff-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_diff libsvn_subr apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_diff libsvn_subr apriconv apr
 
 # ----------------------------------------------------------------------------
 # Tests for libsvn_ra
@@ -1272,11 +1273,11 @@ libs = libsvn_test libsvn_diff libsvn_subr apriconv apr
 [ra-test]
 description = Test a few things in libsvn_ra
 type = exe
-path = subversion/tests/libsvn_ra
-sources = ra-test.c
+path = subversion/tests
+sources = libsvn_ra/ra-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_ra libsvn_ra_svn libsvn_fs libsvn_delta libsvn_subr
-       apriconv apr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_ra libsvn_fs libsvn_delta libsvn_subr
+        apriconv apr
 
 # ----------------------------------------------------------------------------
 # Tests for libsvn_ra_local
@@ -1284,10 +1285,10 @@ libs = libsvn_test libsvn_ra libsvn_ra_svn libsvn_fs libsvn_delta libsvn_subr
 [ra-local-test]
 description = Test a few things in libsvn_ra_local
 type = exe
-path = subversion/tests/libsvn_ra_local
-sources = ra-local-test.c
+path = subversion/tests
+sources = libsvn_ra_local/ra-local-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_ra_local libsvn_ra libsvn_fs libsvn_delta libsvn_subr
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_ra_local libsvn_ra libsvn_fs libsvn_delta libsvn_subr
        apriconv apr
 
 # ----------------------------------------------------------------------------
@@ -1296,63 +1297,63 @@ libs = libsvn_test libsvn_ra_local libsvn_ra libsvn_fs libsvn_delta libsvn_subr
 [conflict-data-test]
 description = Test the storage of tree conflict data
 type = exe
-path = subversion/tests/libsvn_wc
-sources = conflict-data-test.c utils.c
+path = subversion/tests
+sources = libsvn_wc/conflict-data-test.c libsvn_wc/utils.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_client libsvn_test libsvn_wc libsvn_subr apriconv apr
+libs = libsvn_client libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_wc libsvn_subr apriconv apr
 msvc-force-static = yes
 
 [db-test]
 description = Test the wc-ng database subsystem
 type = exe
-path = subversion/tests/libsvn_wc
-sources = db-test.c utils.c
+path = subversion/tests
+sources = libsvn_wc/db-test.c libsvn_wc/utils.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_client libsvn_test libsvn_wc libsvn_subr apriconv apr
+libs = libsvn_client libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_wc libsvn_subr apriconv apr
 msvc-force-static = yes
 
 [pristine-store-test]
 description = Test the wc-ng pristine text storage subsystem
 type = exe
-path = subversion/tests/libsvn_wc
-sources = pristine-store-test.c utils.c
+path = subversion/tests
+sources = libsvn_wc/pristine-store-test.c libsvn_wc/utils.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_client libsvn_test libsvn_wc libsvn_subr apriconv apr
+libs = libsvn_client libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_wc libsvn_subr apriconv apr
 msvc-force-static = yes
 
 [entries-compat-test]
 description = Test backwards compat for the entry interface
 type = exe
-path = subversion/tests/libsvn_wc
-sources = entries-compat.c utils.c
+path = subversion/tests
+sources = libsvn_wc/entries-compat.c libsvn_wc/utils.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_client libsvn_test libsvn_wc libsvn_subr apriconv apr
+libs = libsvn_client libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_wc libsvn_subr apriconv apr
 msvc-force-static = yes
 
 [op-depth-test]
 description = Test layered tree changes
 type = exe
-path = subversion/tests/libsvn_wc
-sources = op-depth-test.c utils.c
+path = subversion/tests
+sources = libsvn_wc/op-depth-test.c libsvn_wc/utils.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_client libsvn_test libsvn_wc libsvn_subr apriconv apr
+libs = libsvn_client libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_wc libsvn_subr apriconv apr
 msvc-force-static = yes
 
 [wc-queries-test]
 description = Test Sqlite query evaluation
 type = exe
-path = subversion/tests/libsvn_wc
-sources = wc-queries-test.c ../../libsvn_subr/sqlite3wrapper.c
+path = subversion/tests
+sources = libsvn_wc/wc-queries-test.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_test libsvn_subr apriconv apr sqlite
+libs = libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_subr apriconv apr sqlite
 
 [wc-test]
 description = Test the main WC API functions
 type = exe
-path = subversion/tests/libsvn_wc
-sources = wc-test.c utils.c
+path = subversion/tests
+sources = libsvn_wc/wc-test.c libsvn_wc/utils.c svn_test_fs.c svn_test_main.c
 install = test
-libs = libsvn_client libsvn_test libsvn_wc libsvn_subr apriconv apr
+libs = libsvn_client libsvn_repos libsvn_fs libsvn_delta libsvn_subr aprutil apriconv apr libsvn_wc libsvn_subr apriconv apr
 msvc-force-static = yes
 
 # ----------------------------------------------------------------------------
